guides/source/threading_and_code_execution.md

```diff
@@ -5,65 +5,144 @@ Threading and Code Execution in Rails
 
-Executor
---------
+Rails' in-built concurrency will cover the day-to-day needs of many application
+developers, and ensure applications remain generally performant.
 
-The Rails Executor separates application code from framework code: any time the
-framework invokes code you've written in your application, it will be wrapped by
-the Executor.
+NOTE: You can read more about how to configure Rails' in-built concurrency in
+the [Framework Behavior](#framework-behavior) section.
 
-The Executor consists of two callbacks: `to_run` and `to_complete`. The Run
-callback is called before the application code, and the Complete callback is
-called after.
+### `CurrentAttributes` and Threading
 
-### Default Callbacks
+The
+[`ActiveSupport::CurrentAttributes`](https://edgeapi.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html)
+class is a special class in Rails that helps you manage temporary data for each
+request in your app, and helps make sure this data is available to the whole
+system. It keeps this data separate for every request (even if there are
+multiple threads running) and makes sure the data is cleaned up automatically
+when the request is done.
 
-In a default Rails application, the Executor callbacks are used to:
+You can think of this class as a place to store data that you need to access
+anywhere in your app without having to pass it around in your code.
+
+To use the `Current` class to store data, first you need to create a file as
+below, with `attribute` values for the attributes and models whose values you
+would like to access throughout your application. You can also define a method
+(e.g the `user` method below) which, when called, will contain set values:
+
+```ruby
+# app/models/current.rb
+class Current < ActiveSupport::CurrentAttributes
+  attribute :account, :user
+
+  resets { Time.zone = nil }
+
+  def user=(user)
+    super
+    self.account = user.account
+    Time.zone    = user.time_zone
+  end
+end
+```
+
+You will now have access to `Current.user` elsewhere in your application. For
+example, when authenticating a user:
+
+```ruby
+# app/controllers/concerns/authentication.rb
+module Authentication
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :authenticate
+  end
+
+  private
+    def authenticate
+      if authenticated_user = User.find_by(id: cookies.encrypted[:user_id])
+        Current.user = authenticated_user
+      else
+        redirect_to new_session_url
+      end
+    end
+end
+```
+
+WARNING: It’s easy to put too many attributes in the `Current` class and tangle
+your model as a result. Current should only be used for a few, top-level
+globals, like account, user, and request details.
+
+### Isolated Execution State
+
+The `active_support.isolation_level` value in your `configuration.rb` file
+provides you the option to define where Rails internal state should be stored
+while tasks are run. If you use a fiber-based server or job processor (e.g.
```

Comment:
Would it be in the scope of this guide to explain what a "fiber-based server" is? And what threads are for that matter? If so, something along these line can be added, somewhere in this guide:

_On a single computer, multiple processes can run. Within a single process, multiple threads can run. Within a single thread, multiple fibers can run._